awc/src/ws.rs

```diff
@@ -233,6 +233,8 @@ impl WebsocketsRequest {
             return Either::A(err(InvalidUrl::UnknownScheme.into()));
         }
 
+        self.head.headers.insert(header::HOST, HeaderValue::from_str(uri.host().unwrap()).unwrap());
```

Comment:
you need to check if header is set already

Reply:
Updated.